tests/baselines/reference/mappedTypeRecursiveInference.types

```diff
@@ -4,7 +4,7 @@
 Assignability cache: 5,000
 Type Count: 10,000
 Instantiation count: 250,000
-Symbol count: 100,000
+Symbol count: 250,000
```

Comment:
...That's a big bump. Is this just caused by the extra `reportRelationError` call expanding the recursive type way more? Still feels like a lot more symbols.

Reply:
This isn't necessarily something that aughta get fixed in this PR, but understanding where _so many symbols_ popped in from could be useful. It might indicate that we're not caching something we aughta be.

Reply:
> Is this just caused by the extra `reportRelationError` call expanding the recursive type way more?

I believe so. Since this test has an infinitely recurring type, it didn't seem like a great reason for concern to me, but I could be wrong.